docs/data/data-grid/aggregation/aggregation.md

```diff
@@ -4,17 +4,250 @@ title: Data Grid - Aggregation
 
 # Data Grid - Aggregation [<span class="plan-premium"></span>](https://mui.com/store/items/mui-x-premium/)
 
-<p class="description">Apply aggregation function to populate the group row with values.</p>
+<p class="description">Use aggregation functions to combine your row values.</p>
+
+You can aggregate rows through the grid interface by opening the column menu and selecting from the items under **Aggregation**.
+
+The aggregated values will be rendered in a footer row at the bottom of the grid.
 
 :::warning
-This feature isn't implemented yet. It's coming.
+The footer row will be pinned at the bottom of the grid once [#1251](https://github.com/mui/mui-x/issues/1251) is ready.
+:::
+
+{{"demo": "AggregationInitialState.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Pass aggregation to the grid
+
+### Structure of the model
+
+The aggregation model is an object.
+The keys correspond to the columns, and the values are the name of the aggregation function to use.
+The values contain the name of the functions that define the aggregation behavior, and where the aggregated values are displayed.
+
+### Initialize aggregation
+
+To initialize aggregation without controlling its state, provide the model to the `initialState` prop.
+
+{{"demo": "AggregationInitialState.js", "bg": "inline"}}
+
+### Controlled aggregation
+
+Use the `aggregationModel` prop to control aggregation passed to the grid.
+
+You can use the `onAggregationModelChange` prop to listen to changes to aggregation and update the prop accordingly.
+
+{{"demo": "AggregationControlled.js", "bg": "inline"}}
+
+## Disable aggregation
+
+### For all columns
+
+You can disable aggregation by setting the `disableAggregation` prop to true.
+
+It will disable all features related to aggregation, even if a model is provided.
+
+{{"demo": "AggregationDisabled.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### For some columns
+
+In case you need to disable aggregation on specific column(s), set the `aggregable` property on the respective column definition (`GridColDef`) to `false`.
+In the example below, the `title` and `year` columns are blocked from being aggregated:
+
+{{"demo": "AggregationColDefAggregable.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Usage with row grouping
+
+When the row grouping is enabled, the aggregated values will be displayed
+
+1. On the top-level footer - like without row grouping, the grid will add a top-level footer to aggregate all the rows
```

Comment:
"Like without" is a strange phrase. Is this a typo? Should it be "Like with" / "as with"?

```suggestion
1. On the top-level footer - as with row grouping, the grid will add a top-level footer to aggregate all the rows
```

Reply:
How about this? (Also rearranging the clauses the put the more important part first)

```suggestion
1. On the top-level footer - the grid will add a top-level footer to aggregate all the rows, as it would with a flat row list
```